mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVMPass.cpp

```diff
@@ -83,21 +83,26 @@ void LowerVectorToLLVMPass::runOnOperation() {
   // Convert to the LLVM IR dialect.
   LowerToLLVMOptions options(&getContext());
   options.useOpaquePointers = useOpaquePointers;
-  LLVMTypeConverter converter(&getContext(), options);
+
+  LLVMTypeConverter *converter;
+  if (armSME)
+    converter = new arm_sme::ArmSMETypeConverter(&getContext(), options);
+  else
+    converter = new LLVMTypeConverter(&getContext(), options);
```

Comment:
This is a memory leak. You should at the very least use `std::unique_ptr<LLVMTypeConverter>` here and use `make_unique` in the logic.

Reply:
ah, thanks for heads up! I'll push a fix soon